src/app/clusters/power-source-configuration-server/power-source-configuration-server.cpp

```diff
@@ -61,9 +70,23 @@ CHIP_ERROR PowerSourceConfigurationAttrAccess::Read(const ConcreteReadAttributeP
     {
     case Sources::Id:
         err = aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
+            uint16_t orderEpPair[kMaxPowerSources * 2];
```

Comment:
Is there a reason to not make this an array of `std::pair` with the right types in it?  That would make the sort compare much clearer, make the iteration when encoding a lot clearer, etc.